apps/ecommerce/frontend/src/helpers/getResourceProviderAndType.ts

```diff
@@ -0,0 +1,12 @@
+import { ExternalResourceLink } from '../types';
+
+const getResourceProviderAndType = (resource?: ExternalResourceLink) => {
+  const [resourceProvider, resourceType] = resource?.sys?.linkType?.split(':') || [];
```

Comment:
incase the linktype is undefined, the output will be [undefined, undefined]. We could also have these values be an error state too like `No Provider` and `No Resource Type`

Reply:
I think you don't need to make the parameter optional. Personally, I think utils functions should be pure functions without all this error handling.

Basically it should be 

```
const getResourceProviderAndType = (resource: ExternalResourceLink) => {
  const [resourceProvider, resourceType] = resource.sys.linkType.split(':') 

  return {
    resourceProvider,
    resourceType,
  };
};
```

and you should do some resource error checking in the caller who will have context on how to handle those situations. That way these functions are cut and dry and what they do, and each component can handle their errors as they need to